framework/core/src/Foundation/Concerns/InteractsWithLaravel.php

```diff
@@ -0,0 +1,201 @@
+    {
+        $index = 0;
+
+        while ($index < count($this->terminatingCallbacks)) {
+            $this->call($this->terminatingCallbacks[$index]);
+
+            $index++;
+        }
+    }
+
+    public function basePath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->base, $path);
+    }
+
+    public function publicPath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->public, $path);
+    }
+
+    public function storagePath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->storage, $path);
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function bootstrapPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'bootstrap'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function configPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->storage, 'config'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function databasePath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'database'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function langPath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'lang'),
+            $path
+        );
+    }
+
+    /** Not actually used/has no meaning in Flarum. */
+    public function resourcePath($path = ''): string
+    {
+        return $this->joinPaths(
+            $this->joinPaths($this->paths->base, 'resources'),
+            $path
+        );
+    }
+
+    public function environment(...$environments)
+    {
+        if (count($environments) > 0) {
+            $patterns = is_array($environments[0]) ? $environments[0] : $environments;
+
+            return Str::is($patterns, $this['env']);
+        }
+
+        return $this['env'];
+    }
+
+    public function runningInConsole(): bool
+    {
+        if ($this->isRunningInConsole === null) {
+            $this->isRunningInConsole = \PHP_SAPI === 'cli' || \PHP_SAPI === 'phpdbg';
+        }
+
+        return $this->isRunningInConsole;
+    }
+
+    public function runningUnitTests(): bool
+    {
+        return $this->bound('env') && $this['env'] === 'testing';
+    }
+
+    public function hasDebugModeEnabled(): bool
+    {
+        return $this->config('debug', true);
+    }
+
+    public function maintenanceMode()
+    {
+        // TODO: Implement maintenanceMode() method.
+        return null;
+    }
+
+    public function isDownForMaintenance(): bool
+    {
+        // TODO: Implement isDownForMaintenance() method.
+        return false;
+    }
+
+    public function registerConfiguredProviders()
```

Comment:
What's up with the empty ones?

Reply:
has no actual meaning within our architecture, but have to implement the contract.

Reply:
Are any of the methods / code in this trait used for our needs at all?

Reply:
Also, could we mark all these as deprecated to discourage using them in Flarum code?

Reply:
> Are any of the methods / code in this trait used for our needs at all?

Indirectly, within illuminate components we use, like `runningInUnitTests` or `runningInConsole`, `terminating`. At any moment an Illuminate component could start calling one of these methods.